### DISCUSSION TITLE
Springboot rds iam failed

### DISCUSSION THREAD

**Discussion by rnei00** [aws/aws-advanced-jdbc-wrapper#477] [Q&A] [answered]
Hello,
I have an issue when i'm trying to connected to database with iam on my Spring boot application : 

Driver org.postgresql.Driver claims to not accept jdbcUrl : jdbc:aws-wrapper:postgresql://$DB_HOST:$DB_PORT/$DB_NAME?wrapperPlugins=iam"

I use these dependencies : 
` <dependency>
            <groupId>org.postgresql</groupId>
            <artifactId>postgresql</artifactId>
            <scope>runtime</scope>
        </dependency>
        <dependency>
            <groupId>software.amazon.jdbc</groupId>
            <artifactId>aws-advanced-jdbc-wrapper</artifactId>
            <version>2.1.2</version>
        </dependency>
   <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>rds</artifactId>
            <version>2.20.78</version>
        </dependency>
`

Properties files:
`spring:
   datasource:
    url: "jdbc:aws-wrapper:postgresql://$DB_HOST:$DB_PORT/$DB_NAME?wrapperPlugins=iam"
    username: "test"
    driver-class-name: "org.postgresql.Driver"
 
Some help from you ?
Thanks.

**Comment by aaronchung-bitquill** [comment]
Hi @rnei00, 

Thanks for reaching out and raising this with us. 

In the properties file, the value of `driver-class-name` should be `software.amazon.jdbc.Driver`. 

Could you kindly try that out and let us know if the issue persists?

Thank you!

**Reply by davecramer** [reply, marked as answer]
looks like In the properties file, the value of driver-class-name should be software.amazon.jdbc.Driver. is the solution

**Comment by rnei00** [comment]
Hello @aaronchung-bitquill ,

Yes it's help me thanks.

Now i have this error :org.postgresql.util.PSQLException: FATAL: PAM authentication failed for user "db_user_rds_iam"

I use the user who have the iam role.

Any suggest ?

**Comment by rnei00** [comment]
It's resolved thanks ;)

**Reply by eratio08** [reply]
@rnei00 What was your solution for this?

**Reply by aaronchung-bitquill** [reply]
Hello @eratio08 

The solution was that the value of the `driver-class-name` property should be `software.amazon.jdbc.Driver`.

**Reply by eratio08** [reply]
I'm using the correct driver class name but I'm still seeing the `PAM authentication failed for user...` error. I've worked though https://repost.aws/knowledge-center/aurora-postgresql-connect-iam but I'm still not successful. Using trace logging I can confirm that the driver wrapper get the token from IAM. Some how it still causes the error.

Update: it seems to be realted to the IAM policy. Using `...dbuser:*/*` resolves it but this is not the solution for me. Will do some experiments to figure out why the resource definition is wrong.

**Reply by aaronchung-bitquill** [reply]
Hi @eratio08 

Just in case you haven't seen it, here is the documentation on [how to use  IAM authentication with the AWS JDBC Driver](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheIamAuthenticationPlugin.md#how-do-i-use-iam-with-the-aws-jdbc-driver). 
It also includes what needs to be set up on the database side of things. 

If you still having issues and would like some assistance, please feel free to open an issue. 
Thank you

**Reply by eratio08** [reply]
I read that one as well, the issue was that I was using the db instance identifier instead of the dbi resource id in the IAM policy. I had to take a harder look at [IAM Policy specification for rds-db](fi-daidalos.collection-case-reopened.determine-purchase-limit.dlq).
